### PR TITLE
Implement remote.Delete

### PR DESCRIPTION
**Pull request by mattmoor**
This adds support for deleting images from a registry by either tag or digest.

This also adds `//cmd/deleter` with a simple wrapper of this functionality.

Fixes: #39

### CODE REVIEW

File: v1/remote/delete.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK, http.StatusAccepted:
+		return nil
+	default:
+		return fmt.Errorf("unrecognized status code during DELETE: %v", resp.Status)
```
Comment (jonjohnsonjr) [inline v1/remote/delete.go]: Can we log.Debug resp.Body here? Or add it to the error?

File: v1/remote/transport/scheme.go [hunk 2]
```diff
@@ -24,13 +24,19 @@ import (
 // Detect more complex forms of local references.
 var reLocal = regexp.MustCompile(`.*\.local(?:host)?(?::\d{1,5})?$`)
 
-// scheme returns https scheme for all the endpoints except localhost.
-func scheme(reg name.Registry) string {
+// Detect the loopback IP (127.0.0.1)
+var reLoopback = regexp.MustCompile(regexp.QuoteMeta("127.0.0.1"))
```
Comment (jonjohnsonjr) [inline v1/remote/transport/scheme.go]: Thanks :)

File: v1/remote/delete.go (hunk 1 above)
Comment (mattmoor) [reply v1/remote/delete.go]: Done.

File: v1/remote/delete.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+)
+
+// DeleteOptions are used to expose optional information to guide or
+// control the image deletion.
+type DeleteOptions struct {
+	// TODO(mattmoor): Fail on not found?
+	// TODO(mattmoor): Delete tag and manifest?
+}
+
+// Delete pushes the provided img to the specified image reference.
```
[the remark this answers is not in the document]
Comment (mattmoor) [reply v1/remote/delete.go]: Done.

**Pull request merged**